### PR TITLE
[Modal] - add z-index to fullscreen modal header

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] add `z-index`

## Screenshots

|  Before  |  After  |
|--------|--------|
|fullscreenModalBefore|fullscreenModalAfter|

## Testing in `sage-lib`

On the fullscreen modal variation, scroll overflowed body text to verify that the content falls behind the header

## Testing in `kajabi-products`

1. (**MEDIUM**) Description of the change and its impact with QA as the audience.
   - [ ] Product Product Creation Wizard. (behind FF: `:product_creation_wizard`)

### CODE REVIEW

**Review by pixelflips — APPROVED** [review]
LGTM!

File: packages/sage-assets/lib/stylesheets/components/_modal.scss [hunk 1]
```diff
@@ -146,6 +146,7 @@ $-modal-fullscreen-top-spacing: rem(104px);
 
   .sage-modal--fullscreen & {
     position: fixed;
+    z-index: sage-z-indexes(default, 1);
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/components/_modal.scss:149]: Should this be `sage-z-index`?

**Review by monkeypox8 — CHANGES_REQUESTED** [review]
Small typo in the token use, should be good to go once that's cleared up though!

Something that we might want to look into later is applying `overscroll-behavior` on modals as default behavior. No Safari support yet, but it could be useful to lock the background page from scrolling.

File: packages/sage-assets/lib/stylesheets/components/_modal.scss (hunk 1 above)
Comment (QuintonJason) [reply packages/sage-assets/lib/stylesheets/components/_modal.scss]: @monkeypox8 nice catch. Pushed a fix

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM!

**Pull request merged**